src/REL/d/a/obj/d_a_obj_block_underground.cpp

```diff
@@ -1,6 +1,292 @@
 #include "d/a/obj/d_a_obj_block_underground.h"
 
+#include "c/c_list.h"
+#include "common.h"
+#include "d/a/d_a_player.h"
+#include "d/a/obj/d_a_obj_base.h"
+#include "d/col/bg/d_bg_s.h"
+#include "d/col/cc/d_cc_d.h"
+#include "d/col/cc/d_cc_s.h"
+#include "d/flag/sceneflag_manager.h"
+#include "d/snd/d_snd_small_effect_mgr.h"
+#include "f/f_base.h"
+#include "f/f_list_nd.h"
+#include "m/m_vec.h"
+#include "nw4r/g3d/res/g3d_resfile.h"
+#include "nw4r/g3d/res/g3d_resmdl.h"
+#include "toBeSorted/d_emitter.h"
+
 SPECIAL_ACTOR_PROFILE(OBJ_BLOCK_UNDERGROUND, dAcOBlockUnderground, fProfile::OBJ_BLOCK_UNDERGROUND, 0x13F, 0, 3);
 
-dAcOBlockUnderground::dAcOBlockUnderground() {}
-dAcOBlockUnderground::~dAcOBlockUnderground() {}
+const s32 dAcOBlockUnderground::sSubtypeIdx[12] = {0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 1, 1};
+
+const char *const dAcOBlockUnderground::sOarcName = "MoleBlock";
+
+const char *const dAcOBlockUnderground::sMdlNames[4] = {
+    "MoleBlock",
+    "MoleBlockBomb",
+    "MoleBlockBreak",
+    "MoleBlockBreak",
+};
+
+const char *const dAcOBlockUnderground::sDzbPlcNames[4] = {
+    "MoleBlock_rock",
+    "MoleBlock_rock",
+    "MoleBlock_tuti",
+    "MoleBlock_tuti",
+};
+
+const dCcD_SrcCyl dAcOBlockUnderground::sCylSrc = {
+    /* mObjInf */
+    {/* mObjAt */ {0, 0, {0, 0, 0}, 0, 0, 0, 0, 0, 0},
+     /* mObjTg */ {AT_TYPE_BOMB, 0x11, {0, 8, 0x407}, 0x0, 0x0},
+     /* mObjCo */ {0}},
+    /* mCylInf */
+    {50.f, 100.f}
+};
+
+u32 dAcOBlockUnderground::getCylTgType() {
+    return 0x20020;
```

Comment:
```suggestion
    return AT_TYPE_0x20000 | AT_TYPE_BOMB;
```

where `AT_TYPE_0x20000` probably means the mogma mitts melee attack?